### PR TITLE
ref(api): Adding a tag to track backend api hits

### PR DESCRIPTION
**Pull request by AniketDas-Tekky**
Trying to differentiate frontend and backend API hits to gather some data. This change checks if a request has a bearer token and records a bool tag if it does.

### CODE REVIEW

File: src/sentry/api/base.py [hunk 1]
```diff
@@ -234,6 +234,10 @@ def dispatch(self, request, *args, **kwargs):
                     # setup default access
                     request.access = access.from_request(request)
 
+            if hasattr(self.request.successful_authenticator, "token_name"):
```
Comment (mgaeta) [inline src/sentry/api/base.py:237]:
```python
if getattr(self.request.successful_authenticator, "token_name", None) == "bearer":
```
Also should this be case-sensitive?

Comment (AniketDas-Tekky) [reply src/sentry/api/base.py:237]: The token name is set here: https://github.com/getsentry/sentry/blob/master/src/sentry/api/authentication.py#L184 so it should be alright.

Comment (AniketDas-Tekky) [reply src/sentry/api/base.py:237]: I updated it to directly check against `TokenAuthentication.token_name`

Comment (mgaeta) [reply src/sentry/api/base.py:237]:
```python
if getattr(self.request.successful_authenticator, "token_name", None) == TokenAuthentication.token_name:
  request._metric_tags["backend_request"] = True
```

**Review by mgaeta — CHANGES_REQUESTED** [review]

File: src/sentry/api/base.py (hunk 1 above)
Comment (AniketDas-Tekky) [reply src/sentry/api/base.py]: Whoops, didn't see the `..., None)` initially

**Review by mgaeta — APPROVED** [review]

**Pull request merged**